sqlmesh/core/console.py

```diff
@@ -2027,6 +2042,25 @@ def show_linter_violations(
             self.log_warning(msg)
 
 
+def _compare_df_cells(x: t.Any, y: t.Any) -> bool:
+    """Helper function to compare two cells and returns true if they're not equal, handling array objects."""
+    if x is None or y is None:
+        return x != y
+
+    # Convert any array-like object to list for consistent comparison
+    def to_list(val: t.Any) -> t.Any:
+        return list(val) if isinstance(val, (pd.Series, np.ndarray, list, tuple, set)) else val
```

Comment:
The `set` here looks incorrect– the ordering in sets is non-deterministic, so converting a set into a list could result in inconsistent results when displaying the table diff.

Additional feedback for this: consider factoring this tuple out into a constant, e.g. `LIST_TYPES` or something.